### ISSUE TITLE
TypeHintDeclarationSniff adds object return type

### ISSUE THREAD

**Issue by enumag** [slevomat/coding-standard#217]
I have a method like this:

```php
	/**
	 * @return object
	 */
	public function foo()
	{
            // ...
	}
```

TypeHintDeclarationSniff changes that to

```php
	public function foo(): object
	{
            // ...
	}
```

It is kind of good that it does but not entirely. Even if I have PHP 7.2 locally it does not mean I want my application to require it right away. I'd like to wait until it's a bit more stable so for now I have this in my composer.json:

```
    "require": {
        "php": "^7.1",
    },
    "config": {
        "platform": {
            "php": "7.1"
        }
    },
```

I don't want to disable `TypeHintDeclarationSniff` nor the code `TypeHintDeclarationSniff.MissingReturnTypeHint`. The sniff should be aware of the target PHP version and not fix things it's not supposed to.

**Comment by kukulich** [comment]
Yes, I was lazy, so `object` type hint is implemented based on running PHP version. Feel free to prepare a PR :)

**Comment by ondrejmirtes** [comment]
It can be done the same way enableVoid property is.

**Comment by kukulich** [comment]
Yes but it must be implemented in more sniffs.

**Comment by kukulich** [comment]
I decided not to implement this as new option because it would be very complicated.

**Comment by kukulich** [comment]
Implemented in https://github.com/slevomat/coding-standard/commit/17acab83ae67ddc38c646e1acd1fcbc19a9faf0f

**Comment by enumag** [comment]
@kukulich Thank you!
